alfresco-authentication/backend/app/src/main/kotlin/com/ritense/valtimo/alfrescoauth/ApplicationReadyEventListener.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2015-2024. Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.ritense.valtimo.alfrescoauth
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.ritense.document.domain.event.DocumentDefinitionDeployedEvent
+import com.ritense.document.service.DocumentDefinitionService
+import com.ritense.plugin.domain.PluginConfiguration
+import com.ritense.plugin.service.PluginConfigurationSearchParameters
+import com.ritense.plugin.service.PluginService
+import com.ritense.valtimo.contract.authentication.AuthoritiesConstants
+import com.ritense.valtimo.contract.json.Mapper
+import org.springframework.boot.context.event.ApplicationReadyEvent
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+@Component
+class ApplicationReadyEventListener(
+    private val pluginService: PluginService,
+    private val documentDefinitionService: DocumentDefinitionService,
+) {
+
+    @EventListener(ApplicationReadyEvent::class)
+    fun handleApplicationReady() {
+        val config = createAlfrescoAuthPluginConfiguration()
+    }
+
+    @EventListener(DocumentDefinitionDeployedEvent::class)
+    fun handleDocumentDefinitionDeployed(event: DocumentDefinitionDeployedEvent) {
+        setDocumentDefinitionRole(event)
+    }
+
+    private fun createAlfrescoAuthPluginConfiguration(): PluginConfiguration {
+        val pluginConfig = pluginService.getPluginConfigurations(PluginConfigurationSearchParameters())
+            .firstOrNull { it.title == "Alfresco Auth configuration" }
+
+        if (pluginConfig != null) {
+            return pluginConfig
+        }
+
+        val configurationProperties = """
+            {
+                "clientId": "testClient",
```

Comment:
We should use externalised configuration for this. Putting credentials in code is not allowed by our development policy, so we should not put it there as an 'example' either.

Reply:
Yes, I now use dummy values. Or would you recommend another value? But for sure, to work locally with it I will need a better solution.